### ISSUE TITLE
Issue: Generating documentation gives error:  "Zod type ZodDate is unsupported"

### ISSUE THREAD

**Issue by gmcouto** [RobinTail/express-zod-api#663]
Whenever I try to generate OpenAPI spec using:

```
const yamlString = new OpenAPI({
  routing: routes, // the same routing and config that you use to start the server
  config,
  version: '1',
  title: 'Posterr API',
  serverUrl: 'https://localhost:8080'
}).getSpecAsYaml();
```

I get the following error:
```
Error: Zod type ZodDate is unsupported
    at depictSchema (/Users/gmcouto/code/strider-backend/node_modules/express-zod-api/src/open-api-helpers.ts:652:11)
    at /Users/gmcouto/code/strider-backend/node_modules/express-zod-api/src/open-api-helpers.ts:124:46
    at Array.map (<anonymous>)
    at depictUnion (/Users/gmcouto/code/strider-backend/node_modules/express-zod-api/src/open-api-helpers.ts:124:18)
```

**Comment by RobinTail** [comment]
Yes, you need to use z.dateIn() and z.dateOut() instead, since z.date() is for `Date` object. See "Dealing with dates" in Readme. @gmcouto

**Comment by RobinTail** [comment]
I should add more clear error for this case

**Comment by RobinTail** [comment]
More clear error message for that case will be released in v8.3.1, @gmcouto 
Thank you for reporting.
